bugsnag-android-core/src/main/jni/root_detection.c

```diff
@@ -5,10 +5,88 @@
 extern "C" {
 #endif
 
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+static const char *su_paths[] = {
```

Comment:
>If the native library cannot be loaded then these checks should be performed on the JVM.

It might be good to alter the `RootDetector.kt` class so that if the native library isn't loaded, it falls back to the JVM root binary checks, but otherwise skips them. This would avoid redundant work - as currently the JVM/NDK are doing the same thing twice.

I think this could be done by updating `isRooted()` and also this else branch: https://github.com/bugsnag/bugsnag-android/blob/6106-root-file-detection2/bugsnag-android-core/src/main/java/com/bugsnag/android/RootDetector.kt#L139

Reply:
Isn't it doing this already?

```
    fun isRooted(): Boolean {
        return try {
            checkBuildTags() || checkSuExists() || checkBuildProps() || checkRootBinaries() || nativeCheckRoot()
        } catch (exc: Throwable) {
            logger.w("Root detection failed", exc)
            false
        }
    }
```
Since the native library can be blocked, we'd have to do some redundant checks; otherwise it would be all-or-nothing native library loading.

Reply:
I'd suggest that the implementation should check if `libraryLoaded.get() == false`, and only call `checkRootBinaries()` if the NDK checks can't occur. This would avoid duplicating the same work.